tests/trilinos/elide_zeros.cc

```diff
@@ -0,0 +1,315 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+/*
+ * Test that Trilinos matrices correctly elide zeros when the zero entry does
+ * not exist in the current process' sparsity pattern.
+ *
+ * The underlying issue is something like this: suppose we have the linear
+ * advection equation (unfortunately we have to work in 2D here so that we can
+ * use distributed objects)
+ *
+ * u_t = u_x + u_y.
+ *
+ * Then, for a two cell triangulation
+ *
+ * > 2 ------- 3 6 ------- 7
+ * > |         | |         |
+ * > |    0    | |    1    |
+ * > |         | |         |
+ * > 0 ------- 1 4 ------- 5
+ *
+ * with FE_DGQ(1) elements there should be no coupling between the degrees
+ * of freedom on the zeroth cell and basis functions 5 and 7 on the first cell
+ * with most numerical fluxes since basis functions 5 and 7 are zero on the
+ * 1-3 edge. Hence the sparsity pattern on rank 0 does not contain a (5, 5) or
+ * (7, 7) entry.
+ *
+ * This is troublesome with MeshWorker or any DG solver that tries to
+ * integrate across each face *exactly once*: the neighbor_to_neighbor_flux
+ * matrix across the 1-3 face could contain all zeros (and will certainly be
+ * zero for rows or columns corresponding to basis functions 5 and 7). Hence I
+ * currently get the following message when running this test on master in
+ * debug mode:
+ *
```

Comment:
Would you mind changing the text in the last sentence to something like: "This would previously give the following exception:"